lib/file_items_gateway.rb

```diff
@@ -6,13 +6,25 @@ class FileItemsGateway
   def all
     return [] unless File.exist?(FILE_PATH)
     File.open(FILE_PATH, 'r') do |file|
-      JSON.parse(file.read, symbolize_names: true)
+      item_builder = ItemBuilder::Item.new
+      item_builder.items = JSON.parse(file.read, symbolize_names: true)
+      [item_builder.build]
     end
   end
 
-  def save(items)
+  def save(item)
+    order = []
+    item.each do |part|
+      row_item = {
```

Comment:
Could this read item row instead?

Reply:
if we change to map we will just need the hash without any assignment i think